pip/commands/install.py

```diff
@@ -201,6 +233,30 @@ def run(self, options, args):
         if options.download_dir:
             options.ignore_installed = True
 
+        if options.platform:
+            if options.download_dir:
+                desired_platform = options.platform
+            else:
+                raise CommandError(
+                    "Usage: Cannot use --platform without also using "
+                    "--download option."
+                )
+        else:
+            desired_platform = get_platform()
+
+        if options.supported_interpreter_version:
+            if options.download_dir:
+                desired_interp_versions = [
+                    options.supported_interpreter_version
+                ]
```

Comment:
Added this information in the `help` field for the `--interpreter-version` option.